ironfish/src/syncer.ts

```diff
@@ -438,7 +440,11 @@ export class Syncer {
       return
     }
 
-    await this.addBlock(peer, block)
+    const { block } = await this.addBlock(peer, newBlock)
+
+    if (!peer.sequence || block.header.sequence > peer.sequence) {
+      peer.sequence = block.header.sequence
```

Comment:
should we also check if we need to update the peer's head and work?

Reply:
How would you know what a peers head and work are from a gossip message? They could have added it to a fork and then gossiped it.